src/frontend/requirements/requirements_.js

```diff
@@ -150,50 +150,59 @@ function getTemplateTable(documentId) {
-  var projectId = PropertiesCache('User', 'projectID');
-
-  numOfCells = 0;
-  var reqs = [];
-  // Check if after inserting a section/group we have inserted requirements
-  var req_inserted = false;
-  var section_inserted = false;
-  insertion_array.reverse();
-  for (line of insertion_array) {
-    // If this is a Group
-    if (Array.isArray(line)) {
-       if(section_inserted && !req_inserted){
-         text_to_insert = "No requirements in section";
-         var body = DocumentApp.getActiveDocument().getBody();
-         var cursor = DocumentApp.getActiveDocument().getCursor();
-         var indexCursor = getCursorIndex(body, cursor);
-         var paragraph = body.insertParagraph(indexCursor + 1, text_to_insert);
-       }
-      // Add Specification or Section Name Name
-      last_index = direct_insert(all_data, line[0], line[1], true);
-      if (reqs.length > 0){
-        reqs.reverse();
-        [tableIndex, tableIndex_, numOfCells] = insertRequirementsInSpec_asTable_fromTemplate(projectId, reqs, all_data, null, numOfCells);
-        reqs = [];
-        req_inserted = true;
-      }
-      section_inserted = true;
-      req_inserted = false;
-    }
-    else {
-      reqs.push(line);
-    }
-  }
+  //console.log("insert_spec_or_group_using_template", insertion_array, all_data);
+
+  try {
+    CacheService.getScriptCache().remove('templateTableData');
+    CacheService.getScriptCache().remove('templateTableCellAttributes');
+    let projectId = PropertiesCache('User', 'projectID');
 
-  // Insert Single Requirement
-  [tableIndex, tableIndex_, numOfCells] = insertRequirementsInSpec_asTable_fromTemplate(projectId, reqs, all_data, null, numOfCells);
-  reqs = []
+    numOfCells = 0;
+    let reqs = [];
 
+    insertion_array.reverse();
 
-  CacheService.getScriptCache().remove('templateTableData')
-  CacheService.getScriptCache().remove('templateTableCellAttributes')
+    for (line of insertion_array) {
+      let req_inserted = false;
+      let section_inserted = false;
 
+      // If this is a Group
+      if (Array.isArray(line)) {
+
+        // Add Specification or Section Name Name
+        last_index = direct_insert(all_data, line[0], line[1], true);
+        if (reqs.length > 0){
+          reqs.reverse();
+          [tableIndex, tableIndex_, numOfCells] = insertRequirementsInSpec_asTable_fromTemplate(projectId, reqs, all_data, null, numOfCells);
+          reqs = [];
+          req_inserted = true;
+        }
+
+        // Check if after inserting a section/group we have inserted requirements
+        if (section_inserted && !req_inserted) {
+          text_to_insert = "No requirements in section";
+          let body = DocumentApp.getActiveDocument().getBody();
+          let cursor = DocumentApp.getActiveDocument().getCursor();
+          let indexCursor = getCursorIndex(body, cursor);
+          let paragraph = body.insertParagraph(indexCursor + 1, text_to_insert);
+        }
+        section_inserted = true;
```

Comment:
This boolean needs to go on line 173 before the if. That is when the section has been inserted and where you can measure after in 182 if it has been inserted